Add incubator/vaultingkube chart

jody-frankowski opened the pull request:
Add a chart for [vaultingkube](https://github.com/sunshinekitty/vaultingkube).

Ping @sunshinekitty

jody-frankowski committed:
Add incubator/vaultingkube chart

unguiculus commented:
/assign

unguiculus commented:
/ok-to-test

unguiculus commented:
Excellent. Thank you.

unguiculus commented on incubator/vaultingkube/templates/deployment.yaml:
```diff
@@ -0,0 +1,56 @@
+apiVersion: apps/v1
```
apps/v1beta2

jody-frankowski replied:
Unless I'm mistaken, I understood that this repository aims to support the latest stable version of k8s, and I understood from the reference documentation that the api version of `Deployment` is `v1` https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.9/#deployment-v1-apps
Could you please explain why i'm wrong :)

jody-frankowski replied:
As a side note, I have only seen the idea of supporting the latest stable version of k8s api in an issue and not in any docs (the docs only say `Do not leverage alpha features of Kubernetes` which does not precise which stable versions of k8s should be supported).

unguiculus replied:
We official support the current and previous minor versions. CI is not at 1.9 yet either.

See https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md#compatibility

jody-frankowski replied:
Done :)

jody-frankowski committed:
Downgrade deployment's apiVersion to v1beta2

In order to be compatible with both k8s 1.8 and 1.9 we need to use deployment's
apiVersion to at most v1beta2.

timstoop commented:
@jody-frankowski Could you take a look at the errors produced by the e2e tests, please?

jody-frankowski commented:
@timstoop Well the chart is expected to fail if the user doesn't provide a certain number of values. While this chart could depend on the Vault chart and configure a Vault if needed, I don't really think it's a good idea, and I don't think too many people would expect this chart to do this.
I tried to follow [this contribution guideline](https://github.com/kubernetes/charts/blame/master/CONTRIBUTING.md#L37)

unguiculus commented:
@jody-frankowski The chart can't be merged if it doesn't pass CI. You may provide special values files for testing if that helps.

See https://github.com/kubernetes/charts/blob/master/test/README.md#providing-custom-test-values

jody-frankowski committed:
Do not deploy unless all the required values are provided

jody-frankowski commented:
@unguiculus  I reproduced what the [minecrat chart did](https://github.com/kubernetes/charts/blob/3bc3476d81095ab4c59a037770586bfb773e13ba/stable/minecraft/templates/deployment.yaml#L1) because until the ci lets me use another chart for the tests, I won't be able to produce a `Running` pod even with the tests ci values.

unguiculus commented:
/lgtm

sunshinekitty commented:
Thanks @jody-frankowski - I'll get documentation on vaultingkube updated on using this chart.
